File: TestApp/src/top/Q34.java
public class Q34 {
    public int[] searchRange(int[] nums, int target) {
        int [] res = new int[]{-1,-1};
        if(nums==null || nums.length==0)
            return res;
        int i = 0;
        int j = nums.length-1;
        while(i<j){
            int mid = (i+j)/2;
            if(nums[mid]<target)
                i = mid+1;
            else
                j = mid;
        }
        if(nums[i]!=target)
            return res;
        res[0] = i;
        j = nums.length-1;
        while(i<j)
        {
            int mid = (i+j)/2 +1;
            if(nums[mid]>target)
                j = mid-1;
            else
                i = mid;
        }
        if(nums[i]!=target)
            res[1] = res[0];
        else
            res[1] = i;
        return res;
        
    }
}

File: TestApp/src/swift/Q34.swift
class Q34 {
    func searchRange(_ nums: [Int], _ target: Int) -> [Int] {
        var res :[Int] = [-1,-1]
        if nums == nil || nums.isEmpty{
            return res
        }
        var i :Int = 0
        var j :Int = nums.count-1
        while i<j{
            let mid:Int = (i+j)/2
            if nums[mid]<target{
                i = mid+1
            }else{
                j = mid
            }
        }
        if nums[i] != target{
            return res
        }
        res[0] = i
        j = nums.count-1
        while i<j{
            let mid = (i+j)/2 + 1
            if nums[mid]>target{
                j = mid-1
            }
            else{
                i = mid
            }
        }
        if nums[i] != target{
            res[1] = res[0]
        }
        else{
            res[1] = i
        }
        return res
    }
}
